.github/workflows/build_artifact.yml

```
@@ -0,0 +1,44 @@
+name: Build Artifact
+
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - v*
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: [ ubuntu-latest ]
```

not running on `self-hosted`?

```suggestion
    runs-on: [ self-hosted, ubuntu-latest ]
```

---

please see my answer below